src/library/Table/Table.js

```diff
@@ -0,0 +1,220 @@
+/* @flow */
+import React, { Component } from 'react';
+import createReactContext, { type Context } from 'create-react-context';
+import { createStyledComponent } from '../styles';
+import { generateId } from '../utils';
+import DataRow from './DataRow';
+import HeaderRow from './HeaderRow';
+import TableBody from './TableBody';
+import TableHeader from './TableHeader';
+import TableTitle from './TableTitle';
+
+type Props = {
+  /** Column definitions ([see example for more details](#column-def)) */
+  columns?: Columns,
+  /** Row data ([see example for more details](#basic)) */
+  data: Array<Object>,
+  /**
+   * Disable the scrolling behavior when Table's width exceeds that of its
+   * container
+   */
+  disableScrollOnOverflow?: boolean,
+  /** Visually hide Table's header */
+  hideHeader?: boolean,
+  /** Render Table with high-contrast styles */
+  highContrast?: boolean,
+  /**
+   * Specifies a key in the row data that gives a row its unique identity.
+   * See the [React docs](https://reactjs.org/docs/lists-and-keys.html#keys).
+   */
+  rowKey?: string,
+  /** Renders Table with alternating row stripes */
+  striped?: boolean,
+  /** Title for Table */
+  title?: React$Node,
+  /** Available title styles (see [Text](/components/text)) */
+  titleAppearance?: 'h1' | 'h2' | 'h3' | 'h4' | 'h5' | 'h6',
+  /** Available title elements (see [Text](/components/text)) */
+  titleElement?: 'h1' | 'h2' | 'h3' | 'h4' | 'h5' | 'h6',
+  /** Amount of vertical padding in Table's cells */
+  verticalSpace: 'default' | 'spacious'
+};
+
+type State = {
+  scrollable: boolean
+};
+
+type Appearance = {
+  highContrast?: boolean,
+  striped?: boolean,
+  verticalSpace?: 'default' | 'spacious'
+};
+export type Columns = Array<Column>;
+// See columnDef example for descriptions
+type Column = {
+  content: React$Node,
+  label?: string,
+  maxWidth?: number | string,
+  minWidth?: number | string,
+  name: string,
+  primary?: boolean,
+  textAlign?: 'start' | 'end' | 'center' | 'justify',
+  width?: number | string
+};
+export type Row = Object;
+export type Rows = Array<Row>;
+
+export const componentTheme = (baseTheme: Object) => ({
+  Table_boxShadow_focus: `0 0 0 1px ${baseTheme.borderColor_theme_focus}`,
+
+  ...baseTheme
+});
+
+const styles = {
+  table: {
+    borderCollapse: 'collapse',
+    borderSpacing: 0,
+    width: '100%'
+  },
+  overflowContainer: ({ theme: baseTheme }) => {
```

Comment:
The styles here cause a flash of the focus box-shadow when content inside the table is focused. Since this is a non-interactive iteration of Table, I thought it was OK to wait to solve that particular issue until Table is interactive.

Reply:
FYI, we decided to punt on this for now.